recipes/llvm-core/all/conanfile.py

```diff
@@ -69,361 +76,265 @@ class LLVMCoreConan(ConanFile):
+            )
+
+        if self._is_windows:
+            if self.options.shared:  # Shared builds disabled just due to the CI
+                raise ConanInvalidConfiguration('Shared builds not currently supported on Windows')
+
         if self.options.exceptions and not self.options.rtti:
-            message = 'Cannot enable exceptions without rtti support'
-            raise ConanInvalidConfiguration(message)
-        self._supports_compiler()
-        if cross_building(self, skip_x64_x86=True):
-            raise ConanInvalidConfiguration('Cross-building not implemented')
+            raise ConanInvalidConfiguration('Cannot enable exceptions without rtti support')
 
     def source(self):
-        get(self, **self.conan_data["sources"][self.version], strip_root=True,
-            destination=self._source_subfolder)
-        self._patch_sources()
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self, generator="Ninja")
+        # https://releases.llvm.org/12.0.0/docs/CMake.html
+        # https://releases.llvm.org/13.0.0/docs/CMake.html
+        cmake_definitions = {
+            'LLVM_TARGETS_TO_BUILD': self.options.targets,
+            # See comment below on LLVM shared library builds
+            'LLVM_BUILD_LLVM_DYLIB': self.options.shared,
+            'LLVM_LINK_LLVM_DYLIB': self.options.shared,
+            'LLVM_DYLIB_COMPONENTS': self.options.components,
+            'LLVM_ABI_BREAKING_CHECKS': 'WITH_ASSERTS',
+            'LLVM_INCLUDE_TOOLS': True,
+            'LLVM_INCLUDE_EXAMPLES': False,
+            'LLVM_INCLUDE_TESTS': False,
+            'LLVM_ENABLE_IDE': False,
+            'LLVM_ENABLE_EH': self.options.exceptions,
+            'LLVM_ENABLE_RTTI': self.options.rtti,
+            'LLVM_ENABLE_THREADS': self.options.threads,
+            'LLVM_ENABLE_LTO': self.options.lto,
+            'LLVM_STATIC_LINK_CXX_STDLIB': self.options.static_stdlib,
+            'LLVM_ENABLE_UNWIND_TABLES': self.options.unwind_tables,
+            'LLVM_ENABLE_EXPENSIVE_CHECKS': self.options.expensive_checks,
+            'LLVM_ENABLE_ASSERTIONS': self.settings.build_type,
+            'LLVM_USE_PERF': self.options.use_perf,
+            'LLVM_ENABLE_Z3_SOLVER': self.options.with_z3,
+            'LLVM_ENABLE_FFI': self.options.with_ffi,
+            'LLVM_ENABLE_ZLIB': "FORCE_ON" if self.options.with_zlib else False,
+            'LLVM_ENABLE_LIBXML2': "FORCE_ON" if self.options.with_xml2 else False,
+            'LLVM_ENABLE_TERMINFO': self.options.with_terminfo,
+            'LLVM_RAM_PER_COMPILE_JOB': self.options.ram_per_compile_job,
+            'LLVM_RAM_PER_LINK_JOB': self.options.ram_per_link_job
+        }
+        if is_msvc(self):
+            build_type = str(self.settings.build_type).upper()
+            cmake_definitions['LLVM_USE_CRT_{}'.format(build_type)] = self.settings.compiler.runtime
+
```

Comment:
Just saw your CI issues, beside the patch there is a second opt needed LLVM_USE_SPLIT_DWARF [its available in llvmorg-11.0.0](https://github.com/llvm/llvm-project/blob/llvmorg-11.0.0/llvm/CMakeLists.txt#L497) therefore I guess this snippet should help. [Since llvm 14 it is also in the docs as recommendation for elf platforms.](https://github.com/llvm/llvm-project/blob/llvmorg-14.0.0/llvm/docs/CMake.rst?plain=1#L757-L761)
```suggestion
        # Conan Center Index CI optimization, in build_type=Debug the CI is killing it because of high memory usage:
        # quote: This option reduces link-time memory usage by reducing the debug information the linker needs to resolve.
        # It is recommended for platforms using the ELF object format, like Linux systems, when linker memory usage is too high.
        is_platform_ELF_based = self.settings.os in [
            'Linux', 'Android', 'FreeBSD', 'SunOS', 'AIX', 'Neutrino', 'VxWorks'
        ]
        if is_platform_ELF_based:
            self.output.info(
                f"your platform \"{self.settings.os}\" is using the ELF format, optimizing memory usage during debug build linking.")
            cmake_definitions.update(
                {
                    'LLVM_USE_SPLIT_DWARF': True
                }
            )
            
```

Reply:
If you don't like setting this value for every build we could use it just for CI with this trick: https://github.com/conan-io/conan-center-index/pull/17509#discussion_r1365797505

Reply:
> If you don't like setting this value for every build we could use it just for CI with this trick: [#17509 (comment)](https://github.com/conan-io/conan-center-index/pull/17509#discussion_r1365797505)

It's a good idea - but if I understand it correctly that suggestion is the reverse of what I want - it requires the user to define the config variable to get non-default behaviour. I've made a feature request to get the reverse, that is, have a config variable that can be checked by recipes to determine if they are running on CCI so that they can take action accordingly. https://github.com/conan-io/conan-center-index/issues/23050